notebooks/3.jump-analysis/nbconverted/3.jump_analysis.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import json
+import pathlib
+import sys
+
+import joblib
+import pandas as pd
+from pycytominer.cyto_utils.features import infer_cp_features
+
+# project module imports
+sys.path.append("../../")  # noqa
+from src.utils import check_feature_order  # noqa
+
+# ## Setting up file paths and parameters
+
+# In[2]:
+
+
+# setting up paths
+results_dir = pathlib.Path("../../results")
+data_split_dir = (results_dir / "1.data_splits/").resolve(strict=True)
+jump_data_dir = pathlib.Path("../../data/JUMP_data").resolve(strict=True)
+modeling_dir = pathlib.Path("../../results/2.modeling").resolve(strict=True)
+
+
+# data files
+jump_data_path = (jump_data_dir / "all_plates_normalized_negcon.csv.gz").resolve(
+    strict=True
+)
+multi_class_model_path = (modeling_dir / "multi_class_model.joblib").resolve(
+    strict=True
+)
+feature_col_names = (data_split_dir / "feature_cols.json").resolve(strict=True)
+
+
+# output paths
+jump_analysis_dir = (results_dir / "3.jump_analysis").resolve()
+jump_analysis_dir.mkdir(exist_ok=True)
+
+
+# ## loading files
+
+# In[3]:
+
+
+# loading in the negatlive controled normalized profiles
+jump_df = pd.read_csv(jump_data_path)
+
+# loading json file containing selected feature names
+with open(feature_col_names, mode="r") as infile:
+    cell_injury_cp_feature_cols = json.load(infile)
+
+# loading json file that contains the coder and decoder injury labels
+with open(data_split_dir / "injury_codes.json") as infile:
+    injury_codes = json.load(infile)
+
+# display dataframe and size
+print("JUMP dataset size:", jump_df.shape)
+jump_df.head()
+
+
+# ## Feature alignment
+# In this section, we are identifying the shared features present in both the cell injury and JUMP datasets.
+# Once these features are identified, we update the JUMP dataset to include only those features that are shared between both profiles for our machine learning application.
+
+# First we identify the CellProfiler (CP) features present in the JUMP data.
+# We accomplish this by utilizing `pycytominer`'s  'infer_cp_features()', which helps us identify CP features in the JUMP dataset.
+
+# In[4]:
+
+
+# get compartments
+metadata_prefix = "Metadata_"
+compartments = list(
+    set(
+        [
+            feature_name.split("_")[0]
+            for feature_name in jump_df.columns.tolist()
+            if not feature_name.startswith("Metadata_")
+        ]
+    )
+)
+
+# find CP features in JUMP dataset
+jump_cp_features = infer_cp_features(jump_df, compartments=compartments)
+meta_features = [
+    meta_feature
+    for meta_feature in jump_df.columns.tolist()
+    if meta_feature.startswith(metadata_prefix)
+]
```

Comment:
```suggestion
meta_features = infer_cp_features(jump_df, compartments=compartments, metadata=True)
```

I think this works too